taichi/transforms/type_check.cpp

```diff
@@ -94,6 +96,7 @@ class TypeCheck : public IRVisitor {
   }
 
   void visit(GlobalPtrStmt *stmt) {
+    stmt->ret_type.set_is_pointer(true);
```

Comment:
Great! We now distinguish ptr from i32! But their size doesn't same, also note that global_ptr's addr can overlaps external_ptr's.

Reply:
Yeah. I'm just implementing a temporary solution before we move to a more delicate type system :-)